libs/datalayout/src/main/java/com/linkedin/openhouse/datalayout/generator/OpenHouseDataLayoutStrategyGenerator.java

```diff
@@ -125,4 +130,20 @@ private double estimateComputeGbHr(long rewriteBytes) {
     double rewriteHours = rewriteSeconds / 3600.0;
     return rewriteHours * EXECUTOR_MEMORY_GB + COMPUTE_STARTUP_COST_GB_HR;
   }
+
+  /** Computes the file entropy as the difference of a set of file's target and actual file size. */
+  private double computeEntropy(Dataset<Long> fileSizes) {
+    // If no files available, MSE is 0.
+    if (fileSizes.count() == 0) {
+      return 0;
+    }
+    // Compute the mean-squared error.
+    double mse = 0.0;
+    Iterator<Long> itr = fileSizes.toLocalIterator();
```

Comment:
Is this more efficient to compute in a spark query. Especially when number of files in table grow large to 10s of millions 
Thoughts? CC: @teamurko

Reply:
Should be faster for tables with a large number of files since it would parallelize, but we'd have to pass the target file size into the query in addition to the tableFileName potentially as another function in TableFileStats? Currently the target file size is a property of the strategy and not defined in the Spark app.

Reply:
I'd bet that data_files metadata table will be partitioned, and local iterator will keep memory usage low. In terms of serial calculation, yeah, it could be a little slow since there is fetching of partitions happening kind of one by one, not because the code itself that sums up stuff.

We could rewrite in spark, spark should be able to serialize the target size constant, right? But I'm OK with the current implementation as well.